src/storage/mapping/ExtensionBasedMapper.ts

```diff
@@ -82,7 +83,7 @@ export class ExtensionBasedMapper implements FileIdentifierMapper {
     // Would conflict with how new extensions are stored
     if (/\$\.\w+$/u.test(filePath)) {
       this.logger.warn(`Identifier ${identifier.path} contains a dollar sign before its extension`);
-      throw new UnsupportedHttpError('Identifiers cannot contain a dollar sign before their extension');
+      throw new BadRequestHttpError('Identifiers cannot contain a dollar sign before their extension');
```

Comment:
At this point I'm actually surprised you didn't throw a NotImplementedHttpError here :D. How is this different from for example a SparqlDataAccessor not allowing non-default graphs or a request to a path that is not supported by a router?

Reply:
You're right indeed!

Reply:
There are some more below in MapperUtil if you're planning on changing. It's just that for me it isn't quite clear what exactly the difference is now between a bad request and not implemented. Is bad request only for the http spec or also the solid spec? Does that mean certain 501s will change to 400s if the solid spec is updated? E.g. regarding sparql update only supporting simple inserts/deletes.